MREUnityRuntime/MREUnityRuntimeLib/App/MixedRealityExtensionApp.cs

```diff
@@ -815,8 +830,8 @@ void ProcessActors2(Transform xfrm)
 						Message = trace.Message
 					},
 					Traces = new List<Trace>() { trace },
-					Actors = actors?.Select((actor) => actor.GeneratePatch()) ?? new ActorPatch[] { },
-					Animations = anims?.Select(anim => anim.GeneratePatch()) ?? new AnimationPatch[] { }
+					Actors = actors?.Select((actor) => actor.GeneratePatch()).ToArray() ?? new ActorPatch[] { },
+					Animations = anims?.Select(anim => anim.GeneratePatch()).ToArray() ?? new AnimationPatch[] { }
```

Comment:
From the comments in Payload.cs, it seems that changing these to arrays was necessary due to the change in WebSocket to serialize payloads off-thread. On the flip side, I'll need to play with this change in isolation to understand how it can deserialize off-thread and not touch Unity objects.

---
In reply to: [419657447](https://github.com/microsoft/mixed-reality-extension-unity/pull/147#discussion_r419657447) [](ancestors = 419657447)